stacks-signer/src/runloop.rs

```diff
@@ -540,8 +601,29 @@ impl<C: Coordinator> RunLoop<C> {
                 OperationResult::SignTaproot(_) => {
                     debug!("Received a signature result for a taproot signature. Nothing to broadcast as we currently sign blocks with a FROST signature.");
                 }
-                OperationResult::Dkg(_point) => {
-                    // TODO: cast the aggregate public key for the latest round here
+                OperationResult::Dkg(point) => {
+                    let epoch = self
+                        .stacks_client
+                        .get_node_epoch()
+                        .unwrap_or(EpochId::Epoch30);
+                    if epoch == EpochId::Epoch25 {
+                        match self
+                            .stacks_client
+                            .cast_aggregate_public_key_vote(point.clone())
+                        {
+                            Ok(txid) => {
+                                self.transactions.push(txid);
+                            }
+                            Err(e) => {
+                                warn!("Failed to cast aggregate public key vote: {:?}", e);
+                            }
+                        }
+                    } else if epoch == EpochId::Epoch30 {
+                        // This is done in another PR...
+                        todo!();
```

Comment:
Does this need to be a split behavior? I think casting the DKG vote should work the same in Epochs 2.5 and 3.0

Reply:
In this PR Jacinta is implementing the ability for signers to broadcast through signerDB: https://github.com/stacks-network/stacks-core/pull/4301

Reply:
I think it does. So in epoch 2.5 we have no ability to enforce miners keep our transactions. In epoch 3, we do. we need to braoadcast via traditional means with a tx fee set to some amount to register the first DKG votes. In epoch 3 onwards, we can do 0 fee transactions. HOWEVER, a 0 fee transaction will cause mempool to auto reject it with a fee too low error. @jcnelson has suggested we avoid special casing this transaction and reqorking mempool logic, by just utilizing stacker DB to get the job done and the enforcement via signing to ensure miners include the transactions in the block.